### ISSUE TITLE
Local Cookies Aren't Removed

### ISSUE THREAD

**Issue by Joel889** [Lusito/forget-me-not#42]
Download a page that uses cookies onto your desktop.
Open it from your desktop.
Close tab.
Cookies remain and cannot be cleaned even using manual cleaning.

**Comment by Joel889** [comment]
Is this referring to bug:
https://bugzilla.mozilla.org/show_bug.cgi?id=1440263

Regardless, I know there is a solution, because another addon had a similar issue but resolved it. I'm sure you are familiar with the extension, it ends in auto delete.

**Comment by Lusito** [comment]
Perhaps bug was the wrong label to give this.. since this is more of an enhancement.
Cookies from stored pages are not bound to a domain.. They only have a file path.
Cookies that are not bound to a domain are currently not handled.
It's not like it's hard to solve.. I've just been very busy lately with moving to a new city.. I finally got internet hooked up today in my new apartment, but there are more important issues right now.

**Comment by Joel889** [comment]
I started noticing domainless cookies frequently, even without opening a webpage from the hard drive. I'm sure how these cookies are set and where they are coming from, but I got them.

**Comment by Lusito** [comment]
Domainless cookies that are from local files always have the path set to where the html file was stored.
Try to inspect the cookie details and see if a path has been set for those cookies, that matches some path on your hard-drive.

If it doesn't have  such a path, then it is probably not a cookie related to local html files. More likely is, that either Firefox or some extension sets these cookies for some reason.

**Comment by Lusito** [comment]
The next version will improve the situation a bit: local cookies can then be cleaned by manual cleanup or on startup. This will however act as if all cookies from the users filesystem are from the same domain. There will be a checkbox under rules (default: checked), which allows you to whitelist files from your computer.

Cleanup on tab close will come at a different time, since this would require some rewriting of existing code.

**Comment by practik** [comment]
FYI, this doesn't seem to be working yet in 1.0.5, even though the release notes for 1.0.4. say that it should be.  I tried clearing local cookies manually and on restart, but it didn't work either time.  I have the checkbox to whitelist local files unchecked. Am I doing something wrong?

**Comment by Lusito** [comment]
yes, every tab that doesn't have a domain name is treated as a local page. This is indeed a bit problematic.

**Comment by Lusito** [comment]
Thanks for the investigation

**Comment by Joel889** [comment]
> 
> 
> Well, the only thing I can think of, that might protect these cookies is the rule, that if a domain of the same name is still open (i.e. other local pages), then these cookies are protected. If you are running manual cleanup, you can disable this behavior by unchecking "Protect currently open domains..."

I don't think this working yet.
